plugin/ale.vim

```diff
@@ -89,6 +89,15 @@ let g:ale_lint_on_save = get(g:, 'ale_lint_on_save', 1)
 " This flag can be set to 1 to enable linting when the filetype is changed.
 let g:ale_lint_on_filetype_changed = get(g:, 'ale_lint_on_filetype_changed', 1)
 
+" This flag can be set to 1 to enable fixing when leaving insert mode.
+let g:ale_fix_on_insert_leave = get(g:, 'ale_fix_on_insert_leave', 0)
+
+" This flag can be set to 0 to disable fixing when the buffer is entered.
+let g:ale_fix_on_enter = get(g:, 'ale_fix_on_enter', 0)
```

Comment:
I'd remove "fix on enter" completely. A file shouldn't be modified just by looking at it.